components/script/dom/websocket.rs

```diff
@@ -201,20 +201,30 @@ impl WebSocketMethods<crate::DomTypeHolder> for WebSocket {
         protocols: Option<StringOrStringSequence>,
     ) -> Fallible<DomRoot<WebSocket>> {
         // Steps 1-2.
-        let url_record = ServoUrl::parse(&url).or(Err(Error::Syntax))?;
+        let mut url_record = ServoUrl::parse(&url).or(Err(Error::Syntax))?;
 
-        // Step 3.
+        // Step 3-4-5.
```

Comment:
This block handles steps 4-6, not 3-5 of https://websockets.spec.whatwg.org/#dom-websocket-websocket

Reply:
You're right, fixed.